### PR TITLE
Implement sponsor logos

### PR DESCRIPTION
**Pull request by schmijos**
This add a **multipart** file upload for sponsor images. The nesting is the following

    sponsor_image: { file: image_file, sponsor_id: sponsor_id }

### CODE REVIEW

**Comment by dbrgn** [comment]
Thanks! Will try to test this in the coming week.

File: db/schema.rb [hunk 1]
```diff
@@ -10,45 +10,55 @@
...
   end
 
-  create_table "sponsors", force: :cascade do |t|
-    t.string   "name"
-    t.text     "description"
-    t.boolean  "active"
-    t.datetime "created_at",  null: false
-    t.datetime "updated_at",  null: false
+  create_table "sponsor_images", force: :cascade do |t|
+    t.bigint "sponsor_id"
```
Comment (dbrgn) [inline db/schema.rb]: Why is this a bigint, and `sponsor_id` in the `sensors` table is an integer?

File: test/models/sponsor_image_test.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class SponsorImageTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
```
Comment (dbrgn) [inline test/models/sponsor_image_test.rb]: That's not a very thorough test :wink: I'd either write a test or remove this.

**Comment by rnestler** [comment]
@dbrgn @schmijos What is the status of this one?

**Comment by dbrgn** [comment]
@schmijos the questions above are still open, right?

**Comment by rnestler** [comment]
@schmijos Monthly bump :slightly_smiling_face:

File: db/schema.rb (hunk 1 above)
Comment (schmijos) [reply db/schema.rb:48]: Rails migrated their datatypes somewhen. Shouldn't matter because `bigint` is bigger.

**Comment by dbrgn** [comment]
@schmijos the question here is still open: https://github.com/gfroerli/gfroerli-api/pull/39#pullrequestreview-134160471 how can I test this? how should the request be structured?

**Comment by schmijos** [comment]
It works for me using the following:

```
http --form POST :3000/api/sponsor_images Authorization:'Bearer 0123456789ABCDEF' sponsor_image[sponsor_id]=1 sponsor_image[file]@./test/fixtures/files/logo.png
```

**Comment by dbrgn** [comment]
@schmijos thanks, this works!

I can now upload one or more logos for a single sponsor. That doesn't really make sense, right? The sponsor and the logo should be an 1:0-1 relation, unless we want to support multiple logo types (e.g. an SVG, a small logo and a large logo). In that case, the table would need a "type" field and a unique constraint on the `(sponsor_id, type)` pair.

Furthermore, it seems like there's no way to query the logo(s) associated with a specific sponsor, right? In that case it's not all that useful :slightly_smiling_face: 

Can it fulfill the following use cases?

- As an API admin, I want to add, replace or delete the logo for sponsor X
- As an API consumer, I want to fetch the logo for sponsor X

(Implementing #5 might solve this if we decide that multiple logos per sponsor are desireable.)

**Comment by schmijos** [comment]
This solution is BS anyways. Can I have some persistent storage?

**Comment by dbrgn** [comment]
> But this solution is BS anyways. Can I have some persistent storage?

Implementing this approach [was your choice](https://github.com/gfroerli/api/issues/22#issuecomment-357522544) :wink: But sure, we can also upload the logos manually to some server (e.g. https://files.coredump.ch/) and just add a logo URL field to the sponsor table. I don't think the uploads need to be managed by the API. (Adding sponsors is a manual task anyways.)

**Comment by dbrgn** [comment]
IIRC the consensus was that we'll simply add an URL field for the logo to the sponsor model.

@schmijos this can probably be closed, right?

**Comment by dbrgn** [comment]
Closing in favor of #70.

**Pull request closed without merging**